From 084918ba41acbb80076c64e7338cf73101b1385e Mon Sep 17 00:00:00 2001
From: Abdo Roig-Maranges <abdo.roig@gmail.com>
Date: Tue, 23 Jun 2015 11:13:13 +0200
Subject: [PATCH] core: fix reversed dependency check in unit_check_unneeded

This was introduced by commit be7d9ff730cb88d7c6a8 and breaks
StopWhenUnneeded=true in the presence of a Requisite dependency.
---
 src/core/unit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index 7bb2afc..fac017c 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -1596,7 +1596,7 @@ static void unit_check_unneeded(Unit *u) {
         static const UnitDependency needed_dependencies[] = {
                 UNIT_REQUIRED_BY,
                 UNIT_REQUIRED_BY_OVERRIDABLE,
-                UNIT_REQUISITE,
+                UNIT_REQUISITE_OF,
                 UNIT_REQUISITE_OF_OVERRIDABLE,
                 UNIT_WANTED_BY,
                 UNIT_BOUND_BY,
-- 
2.4.5

